common/achievement/metrics.ts

```diff
@@ -0,0 +1,34 @@
+import 'reflect-metadata';
+// ↑ Needed by typegraphql: https://typegraphql.com/docs/installation.html
+import { ActionID } from '../notification/actions';
+import { Metric } from './types';
+
+export const metricByName: Map<string, Metric> = new Map();
+const metricsByAction: Map<ActionID, Metric[]> = new Map();
```

Comment:
Why is there a getter for only one of them?

Reply:
metricByName is not necessary to get something. MetricByName is used to register all metrics found in the metric.ts. It is checked, if the metric was already registered or not.